src/lib/math/pcurves/pcurves_secp256r1/pcurves_secp256r1.cpp

```diff
@@ -6,15 +6,115 @@
 
 #include <botan/internal/pcurves_instance.h>
 
+#include <botan/internal/pcurves_nist.h>
 #include <botan/internal/pcurves_wrap.h>
 
 namespace Botan::PCurve {
 
 namespace {
 
-// clang-format off
+template <typename Params>
+class Secp256r1Rep final {
+   public:
+      static constexpr auto P = Params::P;
+      static constexpr size_t N = Params::N;
+      typedef typename Params::W W;
+
+      // Adds 4 * P-256 to prevent underflow
+      static constexpr auto P256_4 =
+         hex_to_words<uint32_t>("0x3fffffffc00000004000000000000000000000003fffffffffffffffffffffffc");
```

Comment:
If you decide to adopt my suggestion, it would be worthwhile to replace this by a constexpr multiplication to save a magic string/number, if we have that somewhere already.